src/common/validation.ts

```diff
@@ -0,0 +1,34 @@
+import {Identity} from 'openfin/_v2/main';
+
+import {FDC3Error, IdentityError} from './errors';
+
+/**
+ * Validates the provided identity and returns an identity stripped of any extraneous properties
+ */
+export function parseIdentity(identity: Identity): Identity {
+    validateIdentityIsWellFormed(identity);
+
+    return {uuid: identity.uuid, name: identity.name || identity.uuid};
+}
+
+/**
+ * Checks that the provided identity adheres to the `Identity` interface
+ */
+export function validateIdentityIsWellFormed(identity: Identity): void {
+    let error = false;
+
+    if (identity === null || typeof identity !== 'object') {
+        error = true;
+    } else {
+        const uuidCheck = typeof identity.uuid === 'string';
+        const nameCheck = !identity.name || typeof identity.name === 'string';
+
+        if (!uuidCheck || !nameCheck) {
+            error = true;
+        }
+    }
+
+    if (error) {
+        throw new FDC3Error(IdentityError.InvalidIdentity, `Invalid identity: ${JSON.stringify(identity)}`);
```

Comment:
Yep, will change

Reply:
Actually, we're calling this precisely in the cases where identity has been deserialized, so don't feel like it's necessary

Reply:
Is that true? Looks like this is also called from the client - before anything is sent over the channel.

This is called by `parseIdentity`, which is called by any client API that expects an `Identity` object. That means this could potentially be called with anything?..

Reply:
Oh, yes, no, no it is not true! Fixed